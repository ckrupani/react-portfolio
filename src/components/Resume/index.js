import React from "react";

function Resume({ title }) {
    return (
        <div>
            <h1>{title}</h1>
            <div className="my-2">
                <span>
                    Download my{" "}
                    <a
                        data-testid="link"
                        href="https://au.linkedin.com/in/chirag-rupani-3309a835"
                        target="_blank"
                        rel="noreferrer"
                    >
                        resume
                    </a>
                </span>
            </div>

            <div className="my-2">
                <h5>Frontend Technologies</h5>
                <ul>
                    <li className="light-color">React</li>
                    <li className="light-color">Redux</li>
                    <li className="light-color">Reselect</li>
                    <li className="light-color">Jest</li>
                </ul>
                <h5>Backend Technologies</h5>
                <ul>
                    <li className="light-color">Java</li>
                    <li className="light-color">NodeJs</li>
                </ul>
            </div>
        </div>
    );
}

export default Resume;
