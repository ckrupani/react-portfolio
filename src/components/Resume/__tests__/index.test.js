import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Resume from "..";

afterEach(cleanup);

describe("Resume component", () => {
    it("renders", () => {
        render(<Resume title={"Resume"} />);
    });

    it("matches snapshot DOM node structure", () => {
        const { asFragment } = render(<Resume title={"Resume"} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("inserts the link to download the resume", () => {
        render(<Resume title={"Resume"} />);

        expect(screen.getByTestId("link")).toHaveTextContent("resume");
    });
});
