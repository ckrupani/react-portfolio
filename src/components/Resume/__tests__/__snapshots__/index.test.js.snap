// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Resume
    </h1>
    <div
      class="my-2"
    >
      <span>
        Download my 
        <a
          data-testid="link"
          href="https://au.linkedin.com/in/chirag-rupani-3309a835"
          rel="noreferrer"
          target="_blank"
        >
          resume
        </a>
      </span>
    </div>
    <div
      class="my-2"
    >
      <h5>
        Frontend Technologies
      </h5>
      <ul>
        <li
          class="light-color"
        >
          React
        </li>
        <li
          class="light-color"
        >
          Redux
        </li>
        <li
          class="light-color"
        >
          Reselect
        </li>
        <li
          class="light-color"
        >
          Jest
        </li>
      </ul>
      <h5>
        Backend Technologies
      </h5>
      <ul>
        <li
          class="light-color"
        >
          Java
        </li>
        <li
          class="light-color"
        >
          NodeJs
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
